lib/active_merchant/billing/gateways/mit.rb

```diff
@@ -0,0 +1,215 @@
+require 'json'
+require 'openssl'
+require 'digest'
+require 'base64'
+
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    class MitGateway < Gateway
+      self.test_url = 'https://wpy.mitec.com.mx/ModuloUtilWS/activeCDP.htm'
+      self.live_url = 'https://wpy.mitec.com.mx/ModuloUtilWS/activeCDP.htm'
+
+      self.supported_countries = ['MX']
+      self.default_currency = 'MXN'
+
+      self.supported_cardtypes = %i[visa master]
+
+      self.homepage_url = 'http://www.centrodepagos.com.mx/'
+      self.display_name = 'MIT Centro de pagos'
+
+      self.money_format = :dollars
+
+      def initialize(options = {})
+        requires!(options, :id_comercio, :user, :apikey, :key_session)
+        super
+      end
+
+      def purchase(money, payment, options = {})
+        authorization = authorize(money, payment, options)
+        return authorization unless authorization.success?
+
+        capture(money, payment, options)
+      end
+
+      def cipher
+        OpenSSL::Cipher::Cipher.new('aes-256-cbc') # ('aes-256-cbc')
+      end
+
+      def cipher_key
+        self.options[:key_session]
+      end
+
+      def decrypt(val, keyinhex)
+        # Splits the first 16 bytes (the IV bytes) in array format
+        unpacked = val.unpack('m')
+        iv_base64 = unpacked[0].bytes.slice(0, 16)
+        # Splits the second bytes (the encrypted text bytes) these would be the
+        # original message
+        full_data = unpacked[0].bytes.slice(16, unpacked[0].bytes.length)
+        # Creates the engine
+        engine = OpenSSL::Cipher::Cipher.new('AES-128-CBC')
```

Comment:
I receive a warning when running my unit tests: `warning: constant OpenSSL::Cipher::Cipher is deprecated`. Recommend using the new style of Cipher creation here: `OpenSSL::Cipher::AES128.new(:CBC)` here and on line 63.